repository/databases/oracle/task.go

```diff
@@ -0,0 +1,190 @@
+package oracle
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"fmt"
+
+	"github.com/blcvn/corev4-explorer/appconfig"
+	"github.com/blcvn/corev4-explorer/entities"
+	"github.com/blcvn/corev4-explorer/types"
+	"github.com/godror/godror"
+)
+
+const (
+	sql_INIT_TABLE_TASK = `CREATE TABLE %s (
+		"ID" NUMBER GENERATED by default on null as IDENTITY,
+		"TYPE" NUMBER,
+		"STATUS" NUMBER,
+		"BLOCK_NUM" NUMBER,
+		"NETWORK_NAME" VARCHAR(255),
+		"CHANNEL_NAME" VARCHAR(255),
+		"TASK_TRANSFORM_ID" NUMBER,
+		"CREATE_AT" TIMESTAMP DEFAULT CURRENT_TIMESTAMP
+	)`
+	sql_INSERT_TASK = `INSERT INTO %s (TYPE, STATUS, BLOCK_NUM, NETWORK_NAME, CHANNEL_NAME)
+	VALUES(:1, :2, :3, :4, :5) RETURNING ID INTO :id `
+
+	sql_UPDATE_TASK_TRANSFORM_ID = `UPDATE %s SET "TASK_TRANSFORM_ID" = :1 WHERE "TYPE" = :2 AND "TASK_TRANSFORM_ID" IS NULL`
+
+	sql_UPDATE_TASK_STATUS = `UPDATE %s SET "STATUS" = :1 WHERE "ID" = :2`
+)
+
+func (odb *OracleDB) InitTableTask(ctx context.Context) error {
+	tableName := appconfig.OnchainTableNameMapping["task"]
+
+	sql1 := fmt.Sprintf(SQL_CHECK_TABLE_EXISTED, tableName)
+	result, err := odb.Client.QueryContext(ctx, sql1, godror.FetchArraySize(1))
+	if err != nil {
+		return err
+	}
+
+	defer result.Close()
+
+	var existed int
+	for result.Next() {
+		if err := result.Scan(&existed); err == nil {
+			err := fmt.Errorf("table %s existed", tableName)
+			odb.logger.Warn(err)
+			return err
+		}
+	}
+
+	sql2 := fmt.Sprintf(sql_INIT_TABLE_TASK, tableName)
+	_, err = odb.Client.ExecContext(ctx, sql2)
```

Comment:
Check the result of RowsAffected to ensure that the expected number of rows are updated or inserted. [important]